server/helpers/validation.ts

```diff
@@ -39,29 +42,122 @@ export const postPUTSchema: ValidationSchema = {
+// Have to play around with class-validation to see how foreign keys could be validated from the body.
+export const requestParamSchema : ValidationSchema = {
+    name: "requestParamSchema",
+    properties: {
+        id: [{
+            type: "isInt"
+        }, {
+            type: "min",
+            constraints: [1]
+        }]
+    }
+}
+
 export const validatePost = (req: Request, res: Response, next: NextFunction) => {
     // Remove any extra fields and turn the plain object in to instance of a Post
     const newpost = plainToClass(Post, req.body, { excludeExtraneousValues: true })
     // See if the newly made Post is valid
     validate(newpost).then(errors => { // errors is an array of validation errors
         if (errors.length > 0) {
-            next(Boom.badRequest("Validation error"));
+            next(validationError);
         } else {
             req.body = newpost;
             next();
         }
     });
 }
 
+export const validateReply = (req: Request, res: Response, next: NextFunction) => {
+    // Remove any extra fields and turn the plain object in to instance of a Post
+    const reply = plainToClass(Reply, req.body, { excludeExtraneousValues: true })
+    // See if the newly made Post is valid
+    validate(reply).then(errors => { // errors is an array of validation errors
+        if (errors.length > 0) {
+            next(validationError);
+        } else {
+            req.body = reply;
+            next();
+        }
+    });
+}
+
+
+interface validPostBody {
+    message?: string,
+    category?: string
+}
+
 // Validating a PUT request can not be done the same way as Post as PUT requires only half the data
 // and even that data can be optional. Thats why the validation in this case is done with a schema
 export const validatePostPUT = (req: Request, res: Response, next: NextFunction) => {
     validate("postPUTSchema", req.body).then(errors => {
         if (errors.length > 0) {
-            next(Boom.badRequest("Validation error"));
+            next(validationError);
+        } else {
+            // Strip unwanted fields.
+            let fields : validPostBody = {};
+            if(req.body.message) {
+                fields.message = req.body.message;
+            }
+            if(req.body.category) {
+                fields.category = req.body.category;
+            }
+            req.body = fields;
+            next();
+        }
+
+    });
+}
+
+interface validReplyBody {
+    message : string
+}
+
+export const validateReplyPUT = (req: Request, res: Response, next: NextFunction) => {
+    validate("replyPUTSchema", req.body).then(errors => {
+        if(errors.length > 0) {
+            next(validationError);
         } else {
+            const valid : validReplyBody = {
+                message: req.body.message
+            }
+            req.body = valid;
             next();
         }
 
     });
 }
+
+interface validParams {
```

Comment:
By convention all interfaces and type are PascalCased so this should be ValidParams

Reply:
For some reason not making an own file for interfaces had me treat them as if they were just functions, this was a good reminder! I'll fix them!